lib/developer_portal/app/controllers/developer_portal/admin/applications_controller.rb

```diff
@@ -113,10 +113,12 @@ def applications
   end
 
   def service
-    @service ||= if not site_account.multiservice?
-                   site_account.services.default
-                 elsif service_id = params[:service_id]
-                   site_account.services.find_by_id_or_system_name(service_id)
+    @service ||= if !multiservice?
+                   services.default
+                 elsif current_account.services_can_create_app_on.count == 1
+                   services.first
```

Comment:
If multiple services switch is disabled and there is no default, then no service will be selected, right? Does it make sense changing to:

```suggestion
    @service ||= if site_account.service_items_in_menu? || current_account.services_can_create_app_on.count == 1
                   services.default || services.first
```

BTW, `Account#service_items_in_menu?` is a horrible method name in a model and it seems no longer to be used anywhere. We might want to enjoy the opportunity to perhaps rename it as `single_service?`.

Reply:
In that case, wouldn't `services.first` be enough?

Reply:
After giving it a though, I'd rather not change the logic of this block. I moved the `service_id` condition to first and renamed the method but otherwise the outcome should be the same. I don't know what will happen if we returned `services.first` so it will redirect to `choose_service` when service is `nil`.

Reply:
We were thinking of removing default service years ago so I think we should just use `services.first`